src/main/resources/testchipip/csrc/cospike_impl.cc

```diff
@@ -327,14 +348,8 @@ int cospike_cosim(long long int cycle,
     loadarch_state_t &ls = dtm->loadarch_state[hartid];
     s->pc  = ls.pc;
     s->prv = ls.prv;
-    s->csrmap[CSR_MSTATUS]->write(s->csrmap[CSR_MSTATUS]->read() | MSTATUS_VS | MSTATUS_XS | MSTATUS_FS);
+    s->csrmap[CSR_MSTATUS]->write(s->csrmap[CSR_MSTATUS]->read() | MSTATUS_VS | MSTATUS_XS | MSTATUS_FS); // TODO: is this unneeded? since the mstatus is reloaded right after
```

Comment:
Any thoughts on this?

Reply:
This is needed. We need to restore the registers even if the mstatus bits say they are clean. So we have to enable VS/XS/FS, write the registers, then restore mstatus.